app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                myItems = multiselectSelected as Dictionary<Channel>;
+                multiselectItems = multiselectItems as Channel[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+            default:
+                myItems = multiselectSelected as Dictionary<DialogOption>;
+                multiselectItems = multiselectItems as DialogOption[];
+                // eslint-disable-next-line guard-for-in
+                for (const index in myItems) {
+                    multiselectItems.push(myItems[index]);
+                }
+                return multiselectItems;
+        }
+    }, [multiselectSelected]);
+
+    // Effects
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, onHandleMultiselectSubmit, [onHandleMultiselectSubmit]);
+
+    useEffect(() => {
+        setLoading(true);
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            getChannels();
+        } else {
+            getDynamicOptionsLocally();
+        }
+
+        setLoading(false);
+    }, []);
+
+    useEffect(() => {
+        let listData: (DataType | UserProfileSection[]) = integrationData;
+        if (term) {
+            listData = searchResults;
+        }
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            listData = createProfilesSections(listData as UserProfile[]);
+        }
+
+        if (!dataSource || dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            listData = (integrationData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().includes(term));
+        }
+
+        setCustomListData(listData);
+    }, [searchResults, integrationData]);
+
+    useEffect(() => {
+        if (!isMultiselect) {
+            return;
+        }
+
+        setButtons(componentId, {
+            rightButtons: [rightButton],
+        });
+    }, [rightButton, componentId]);
+
+    useEffect(() => {
+        const multiselectItems: MultiselectSelectedMap = {};
+
+        if (isMultiselect && selected && !([ViewConstants.DATA_SOURCE_USERS, ViewConstants.DATA_SOURCE_CHANNELS].includes(dataSource))) {
+            selected.forEach((opt) => {
+                multiselectItems[opt.value] = opt;
+            });
+
+            setMultiselectSelected(multiselectItems);
+        }
+    }, [multiselectSelected]);
+
+    // Renders
+    const renderLoading = (): React.ReactElement<any, string> | null => {
```

Comment:
I don't like the `any`. Any chance we can remove it? Maybe the return type can be just `React.ReactNode`?

Reply:
I tried but `renderLoading` wants that.

![Screenshot 2022-11-15 at 12 34 45](https://user-images.githubusercontent.com/488556/201909899-577473c8-392e-450c-929d-116f14217175.png)

Reply:
Looking at the definition, the first argument seems to be the props of the element. Since the element you are returning is not expecting any props, probably the correct type is {}. That being said, what if you directly remove the type of the function?

Reply:
Done